src/nodes.coffee

```diff
@@ -3878,12 +3875,26 @@ exports.While = class While extends Base
         else
           body = Block.wrap [new If @guard, body] if @guard
       body = [].concat @makeCode("\n"), (body.compileToFragments o, LEVEL_TOP), @makeCode("\n#{@tab}")
-    answer = [].concat @makeCode(set + @tab + "while ("), @condition.compileToFragments(o, LEVEL_PAREN),
+    answer = [].concat @makeCode(set + @tab + "while ("), @processedCondition().compileToFragments(o, LEVEL_PAREN),
       @makeCode(") {"), body, @makeCode("}")
     if @returns
       answer.push @makeCode "\n#{@tab}return #{rvar};"
     answer
 
+  processedCondition: ->
+    @processedConditionCache ?= if @invert then @condition.invert() else @condition
+
+  astType: -> 'WhileStatement'
+
+  astProperties: (o) ->
+    return
+      test: @condition.ast o, LEVEL_PAREN
+      body: @body.ast o, LEVEL_TOP
+      guard: @guard?.ast(o) ? null
```

Comment:
`guard`, `inverted`, `postfix` and `loop` are all extensions to Babel's `WhileStatement` AST